File: sdk/typescript/src/api/graphql.ts
import { HTTPClient } from './http-client';

export class GraphQLAPI {
  constructor(private client: HTTPClient) {}

  async execute(query: string, variables?: Record<string, any>): Promise<any> {
    const data = { query };
    if (variables) {
      (data as any).variables = variables;
    }

    return this.client.request<any>({
      method: 'POST',
      endpoint: '/graphql',
      data,
    });
  }
}

File: sdk/rust/src/api/graphql.rs
use crate::client::HTTPClient;
use crate::error::TelecomError;
use crate::types::GraphQLRequest;
use std::collections::HashMap;

pub struct GraphQLAPI {
    client: HTTPClient,
}

impl GraphQLAPI {
    pub fn new(client: HTTPClient) -> Self {
        Self { client }
    }

    pub async fn execute(
        &self,
        query: String,
        variables: Option<HashMap<String, serde_json::Value>>,
    ) -> Result<HashMap<String, serde_json::Value>, TelecomError> {
        let req = GraphQLRequest { query, variables };
        self.client.post("/graphql", Some(&req)).await
    }
}
